subsys/net/ip/tcp2.c

```diff
@@ -606,6 +611,160 @@ static void tcp_out(struct tcp *conn, u8_t flags)
 	tcp_out_ext(conn, flags, NULL /* no data */, conn->seq);
 }
 
+static int tcp_pkt_pull(struct net_pkt *pkt, size_t len)
+{
+	int total = net_pkt_get_len(pkt);
```

Comment:
Calculating the length could be optimized a bit. Instead of each function below doing it itself, we could calculate it once and then just adjust the length if the pull succeeds. What do you think?

Reply:
Sure, done.

Reply:
Everywhere, send_data_total is used. In tcp_pkt_pull() let's keep it as an  extra sanity check.